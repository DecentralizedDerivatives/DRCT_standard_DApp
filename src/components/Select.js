import React from 'react';

export const Select = props => {
  const renderSelectOptions = (key, index) => {
    return (
      <option key={`${index}-${key}`} value={key}>
        {props.options[key]}
      </option>
    );
  };

  if (props && props.options) {
    return (
      <div className="mv3 w-100">
        <div className="b sans-serif pv2 w-100">{props.label}</div>
        <select
          {...props.input}
          className="pa2 input-reset ba b--black-40 w-100"
        >
          <option value="">Select</option>
          {Object.keys(props.options).map(renderSelectOptions)}
        </select>
      </div>
    );
  }
  return <div />;
};

export default Select;
