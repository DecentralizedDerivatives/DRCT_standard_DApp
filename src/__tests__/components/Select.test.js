import Select from '../../components/Select';

describe('<Select />', () => {
  it('renders the component', () => {
    const wrapper = shallow(
      <Select
        meta={{ touched: true, error: 'required' }}
        options={{ option1: 'Option 1' }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders empty component', () => {
    const wrapper = shallow(
      <Select
        meta={{}}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
