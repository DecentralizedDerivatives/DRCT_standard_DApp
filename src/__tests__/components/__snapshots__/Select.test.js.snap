// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> renders empty component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Select
    meta={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Select /> renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Select
    meta={
      Object {
        "error": "required",
        "touched": true,
      }
    }
    options={
      Object {
        "option1": "Option 1",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="b sans-serif pv2 w-100"
        />,
        <select
          className="order-form-dropdown"
        >
          <option
            value=""
          >
            Select
          </option>
          <option
            value="option1"
          >
            Option 1
          </option>
        </select>,
        <span>
          required
        </span>,
      ],
      "className": "mv3 w-100",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "b sans-serif pv2 w-100",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              value=""
            >
              Select
            </option>,
            Array [
              <option
                value="option1"
              >
                Option 1
              </option>,
            ],
          ],
          "className": "order-form-dropdown",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Select",
              "value": "",
            },
            "ref": null,
            "rendered": "Select",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "0-option1",
            "nodeType": "host",
            "props": Object {
              "children": "Option 1",
              "value": "option1",
            },
            "ref": null,
            "rendered": "Option 1",
            "type": "option",
          },
        ],
        "type": "select",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "required",
        },
        "ref": null,
        "rendered": "required",
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="b sans-serif pv2 w-100"
          />,
          <select
            className="order-form-dropdown"
          >
            <option
              value=""
            >
              Select
            </option>
            <option
              value="option1"
            >
              Option 1
            </option>
          </select>,
          <span>
            required
          </span>,
        ],
        "className": "mv3 w-100",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "b sans-serif pv2 w-100",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                value=""
              >
                Select
              </option>,
              Array [
                <option
                  value="option1"
                >
                  Option 1
                </option>,
              ],
            ],
            "className": "order-form-dropdown",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Select",
                "value": "",
              },
              "ref": null,
              "rendered": "Select",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "0-option1",
              "nodeType": "host",
              "props": Object {
                "children": "Option 1",
                "value": "option1",
              },
              "ref": null,
              "rendered": "Option 1",
              "type": "option",
            },
          ],
          "type": "select",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "required",
          },
          "ref": null,
          "rendered": "required",
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
